plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx

```diff
@@ -0,0 +1,264 @@
+  }
+
+  let thead: thead | undefined;
+  let tbody: tbody | undefined;
+  let colgroup: colgroup | undefined;
+  React.Children.forEach(table.props.children, node => {
+    if (node.type === 'thead') {
+      thead = node;
+    } else if (node.type === 'tbody') {
+      tbody = node;
+    } else if (node.type === 'colgroup') {
+      colgroup = node;
+    }
+  });
+  if (!thead || !tbody) {
+    throw new Error('<table> in <StickyWrap> must contain both thead and tbody.');
+  }
+
+  const theadRef = useRef<HTMLTableSectionElement>(null);
+  const scrollHeaderRef = useRef<HTMLDivElement>(null); // fixed header
+  const scrollBodyRef = useRef<HTMLDivElement>(null); // main body
+
+  const { width, height, bodyHeight, columnWidths } = sticky;
+
+  // update scrollable area and header column sizes when mounted
+  useLayoutEffect(() => {
+    if (!height || !theadRef.current) {
+      return;
+    }
+    const bodyThead = theadRef.current;
+    const theadHeight = bodyThead.clientHeight;
+    const tableHeight = (bodyThead.parentNode as HTMLTableElement).clientHeight;
+    const ths = bodyThead.childNodes[0].childNodes as NodeListOf<HTMLTableHeaderCellElement>;
+    const widths = Array.from(ths).map(th => th.clientWidth);
+    const maxHeight = Math.min(height, tableHeight);
+    const newSize = {
+      height: maxHeight,
+      bodyHeight: maxHeight - theadHeight,
+      columnWidths: widths,
+    };
+    setStickyElementSize(newSize);
+  }, [height, setStickyElementSize]);
+
+  if (!columnWidths) {
+    const theadWithRef = React.cloneElement(thead, { ref: theadRef });
+    const bodyTable = React.cloneElement(table, {}, colgroup, theadWithRef, tbody);
+    return <div style={{ height, overflow: 'hidden' }}>{bodyTable}</div>;
+  }
+
+  // if users didn't specify <colgroup>, use computed width
+  colgroup = colgroup || (
+    <colgroup>
+      {columnWidths.map((w, i) => (
+        // eslint-disable-next-line react/no-array-index-key
+        <col key={i} width={w} />
+      ))}
+    </colgroup>
+  );
+
+  const fallbackBodyHeight = bodyHeight || (height && height - DEFAULT_THEAD_HEIGHT);
+  const totalWidth = columnWidths.reduce((a, b) => a + b);
+  const needHorizontalScroll = width && width < totalWidth;
+
+  const tableStyle: CSSProperties = { tableLayout: 'fixed', width: totalWidth };
+  const wrapperStyle: CSSProperties = {
+    width,
+    height,
+    overflow: 'hidden',
+  };
+  const bodyStyle: CSSProperties = {
+    height: fallbackBodyHeight,
+    overflow: 'scroll',
+  };
+  const headerStyle: CSSProperties = {
+    overflow: 'hidden',
+  };
+  const headerTable = React.cloneElement(table, mergeStyleProp(table, tableStyle), colgroup, thead);
+  const bodyTable = React.cloneElement(table, mergeStyleProp(table, tableStyle), colgroup, tbody);
+  const onScroll: UIEventHandler<HTMLDivElement> = e => {
+    if (scrollHeaderRef.current) {
+      scrollHeaderRef.current.scrollLeft = e.currentTarget.scrollLeft;
```

Comment:
Had to manually sync the header's scroll position with the body. Both `jquery.DataTables` and `rc-tables` are doing the same.